packages/admin/src/table/filterbar/filterBarSingleSelect/FilterBarSingleSelect.tsx

```diff
@@ -0,0 +1,87 @@
+import { Check, ChevronDown } from "@comet/admin-icons";
+import { MenuItemProps, Select, SelectProps, WithStyles } from "@material-ui/core";
+import { withStyles } from "@material-ui/styles";
+import * as React from "react";
+
+import { FilterBarSingleSelectClassKey } from "./FilterBarSingleSelect.styles";
+import { styles } from "./FilterBarSingleSelect.styles";
+
+export type MenuItemValues = string | ReadonlyArray<string> | number | undefined;
+
+export interface FilterBarSingleSelectProps<FieldValue extends MenuItemValues> extends SelectProps {
+    value: FieldValue;
+    onChange: (event: React.ChangeEvent<FieldValue> | FieldValue | unknown) => void;
+    children: Array<React.ReactElement<MenuItemProps>>;
+    hideCheck?: boolean;
+}
+
+const SingleSelect = <FieldValue extends MenuItemValues>({
+    value,
+    onChange,
+    hideCheck = false,
+    children,
+    classes,
+    ...props
+}: FilterBarSingleSelectProps<FieldValue> & WithStyles<typeof styles>): React.ReactElement => {
+    let items = children;
+
+    if (!hideCheck) {
+        items = React.Children.map(children, (child: React.ReactElement<MenuItemProps>) => {
+            const { children, value: menuItemValue } = child.props;
+            return React.cloneElement(child, {
+                children: (
+                    <>
+                        {children}
+                        {value === menuItemValue && <Check />}
+                    </>
+                ),
+            });
+        });
+    }
+
+    return (
+        <div className={classes.wrapper}>
+            <Select
+                className={classes.root}
```

Comment:
I think `root` should always be the outermost element... maybe like this:

```suggestion
        <div className={classes.root}>
            <Select
                className={classes.select}
```